Source/Xamarin/Prism.DryIoc.Forms/DryIocExtensions.cs

```diff
@@ -0,0 +1,43 @@
+using DryIoc;
+using Prism.Common;
+using Xamarin.Forms;
+
+namespace Prism.DryIoc
+{
+    /// <summary>
+    /// Extension methods to register a <see cref="Xamarin.Forms.Page"/> with <see cref="PageNavigationRegistry"/> using <see cref="IContainer"/>
+    /// </summary>
+    public static class DryIocExtensions
+    {
+        /// <summary>
+        /// Registers a Page for navigation using a convention based approach, which uses the name of the Type being passed in as the unique name.
+        /// </summary>
+        /// <typeparam name="TPage">The Type of Page to register</typeparam>
+        public static void RegisterTypeForNavigation<TPage>(this IContainer container) where TPage : Page
+        {
+            container.RegisterTypeForNavigation<TPage>(typeof(TPage).FullName);
```

Comment:
Line 83 in `PageNavigationService` use `typeof(ViewModel).FullName`. Shouldn't this be using Name as well, otherwise it won't work